basis/ui/gadgets/scrollers/scrollers.factor

```diff
@@ -96,10 +96,12 @@ M: rect update-scroller swap (scroll>rect) ;
 M: f update-scroller drop (update-scroller) ;
 
 M: scroller layout*
-    [ call-next-method ] [
-        dup follows>>
-        [ update-scroller ] [ >>follows drop ] 2bi
-    ] bi ;
+    {
+      [ call-next-method ]
+      [ dup follows>> [ update-scroller ] [ >>follows drop ] 2bi ]
+      [ [ x>> ] [ y>> ] bi [ forget-pref-dim ] bi@ ]
+      [ call-next-method ]
```

Comment:
Also, why does it need to ``call-next-method`` twice?  Maybe the right way is to not ``forget-pref-dim``, but instead to do:

```[ x>> ] [ y>> ] bi [ relayout-1 ] bi@``

that should trigger layout before the next iteration of the event loop?  (``relayout`` might also work, not sure if it flickers or not).